### DISCUSSION TITLE
STM32F777 controller support availability on zephyr

### DISCUSSION THREAD

**Discussion by enthunilu** [zephyrproject-rtos/zephyr#48688] [General] [answered]
Hi All,

I am exploring STM32 supported controller for my project. And wanted to know, do we have support planned for STM32F777 controller ?
I could not see this controller listed under [https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/st/f7](https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/st/f7)

Is there a workaround by which I can make it enable by using other similar controller dts file.
For example, using stm32f767.dtsi file for stm32f777

Can anyone please help ?

Thanks!!

**Comment by erwango** [comment, marked as answer]
Hi, there is no such plan, but you can look at adding it. Adding a SoC is not a complex task.
See first 2 commits of https://github.com/zephyrproject-rtos/zephyr/pull/41361

**Reply by enthunilu** [reply]
Thank you so much @erwango for your quick response. Surely this will help.
One more question, can you also suggest to enable/add hardware features to STM32F777 what all things I have to do ?
Any link or similar commit will also help.

Thanks Again!!

**Reply by erwango** [reply]
Difference between stm32f777 and already ported stm32f767 is only crypto support.
So in a first step, you can include stm32f767.dtsi in stm32f777.dtsi. And later on, if required you can add crypto like done in https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/st/f4/stm32f417.dtsi

**Reply by enthunilu** [reply]
Thank you so much @erwango :-) 😃
